contrib/federations_tutorial/Federation.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+import os
+import random
+import sys
+import time
+import shutil
+import json
+import util
+from MultiSig import MultiSig
+from BlockSigning import BlockSigning
+from authproxy import AuthServiceProxy, JSONRPCException
+
+ELEMENTS_PATH = "../../src/elementsd"
+GENERATE_KEYS = True
+BLOCK_TIME = 60
+
+def main():
+    # GENERATE KEYS AND SINGBLOCK SCRIPT FOR FEDERATION BLOCK SIGNING
+    num_of_nodes = 3
+    num_of_sigs = 2
+    keys = []
+    signblockarg = ""
+
+    if GENERATE_KEYS:  # generate new signing keys and multisig
+        sig = MultiSig(num_of_nodes, num_of_sigs, ELEMENTS_PATH)
+        sig.generate()
+        keys = sig.privkeys
+        signblockarg = "-signblockscript={}".format(sig.multisig["redeemScript"])
+
+    else:   # use hardcoded keys and multisig
+        keys = ['Kz4aMbbciZrjrNBi4rwwrTemGKbj9qCdVVRKXAvyt1YK1mPFpRG8',
```

Comment:
Perhaps the hardcoded version could be loaded from a file that is generated when using GENERATE_KEYS = True. Just a suggestion, feel free to ignore.